### DISCUSSION TITLE
Superdroplet time series

### DISCUSSION THREAD

**Discussion by jbarr444** [open-atmos/PySDM#1360] [General]
Hi @slayoo,

I was wondering how one might get the history of a superdroplet (its radius time series, for example) as a product of a simulation. Is this possible?

Thanks,
Jason

**Comment by slayoo** [comment]
Hi @jbarr444,

PySDM uses the term `product` to denote (cell/column/grid)-aggregated quantities, while `attribute` is what can be used to analyze the timeseries of particle properties, e.g. radius.

Accessing both can be done through the instance of `Particulator` with `particulator.products` and `particulator.attributes`.
Here you can find one relevant example: https://github.com/open-atmos/PySDM/blob/main/examples/PySDM_examples/Grabowski_and_Pawlowska_2023/figure_1.ipynb

Attributes are being saved here: https://github.com/open-atmos/PySDM/blob/9e153ee743cf1830e28ebfe4530010891523e57e/examples/PySDM_examples/Grabowski_and_Pawlowska_2023/simulation.py#L109-L112

In the Shipway & Hill examples, picking which attributes to save is controlled from here now: https://github.com/open-atmos/PySDM/blob/9e153ee743cf1830e28ebfe4530010891523e57e/examples/PySDM_examples/Shipway_and_Hill_2012/simulation.py#L178-L183
... but it probably would be worth to improve it to be user-controlled and not hardcoded.

Finally, note that there is a question of particle identity preservation upon collisions/breakup, here's a relevant discussion: https://github.com/open-atmos/PySDM/discussions/1361

HTH,
Sylwester

**Reply by jbarr444** [reply]
That helps a lot! Thanks as always.
